File: day1/second.py
with open("input.txt") as file:
    content = file.read()
    lines = content.splitlines()
    lines.append('')
    print(lines)
    current = 0
    result = []
    for line in lines:
        if not line.isdigit():
            result.append(current)
            current = 0
        else:
            n_line = int(line)
            current += n_line
    print(sum(sorted(result, reverse=True)[:3]))

File: day1/second.rb
# frozen_string_literal: true

File.open('input.txt') do |file|
  # @type [String]
  content = file.read
  lines = content.split("\n")
  current = 0
  result = []
  lines.each do |line|
    if line.match(/^(\d)+$/)
      n_line = Integer(line)
      current += n_line
    else
      result.append(current)
      current = 0
    end
  end
  puts (result.sort_by { |x| -x }).take(3).reduce(&:+)
end
